common/stacks/src/main/scala/weco/api/stacks/services/elastic/ElasticItemLookup.scala

```diff
@@ -64,44 +65,77 @@ class ElasticItemLookup(
     }
   }
 
+  private def buildSourceIdentifierSearchRequest(
+    index: Index
+  )(sourceIdentifier: SourceIdentifier) =
+    search(index)
+      .query(
+        boolQuery
+          .must(termQuery(field = "type", value = "Visible"))
+          .should(
+            termQuery(
+              "data.items.id.sourceIdentifier.value",
+              sourceIdentifier.value
+            )
+          )
+      )
+      // TODO: How many things do we need to retrieve here?
+      .size(10)
+
+  val searchRequestOnIndex = buildSourceIdentifierSearchRequest(index)(_)
+
+  def matchItemsOnWorksBySourceIdentifier(
+    works: Seq[Work[Indexed]],
+    sourceIdentifier: SourceIdentifier
+  ) =
+    works
+      .flatMap { _.data.items }
+      .collect {
+        case item @ Item(id @ IdState.Identified(_, _, _), _, _, _)
+            if id.sourceIdentifier == sourceIdentifier =>
+          // This .asInstanceOf[] is a no-op to help the compiler see what
+          // we can see by reading the code.
+          item.asInstanceOf[Item[IdState.Identified]]
+      }
```

Comment:
I'm struggling to wrap my head around these new methods – prefixing the `sourceIdentifier` variables with an `item` or `work` would help me, I think. My brain sees `work` and `sourceIdentifier` next to each other, and instinctively wants to associate the two, even though I think the source identifier is actually for the item?

Reply:
If we remove/replace `bySourceIdentifier` there is no need for this to be its own function, so that kind of solves the problem.